### PR TITLE
PMM-9474 Support PostgreSQL queries for multiple databases in Advisor Checks

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-9474

Link to the Feature Build: SUBMODULES-3106

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- https://github.com/percona-platform/platform/pull/649
- https://github.com/percona-platform/saas/pull/69

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -52,7 +52,7 @@ require (
 	github.com/operator-framework/api v0.17.3
 	github.com/operator-framework/operator-lifecycle-manager v0.23.1
 	github.com/percona-platform/dbaas-api v0.0.0-20230103182808-d79c449a9f4c
-	github.com/percona-platform/saas v0.0.0-20221014123257-4fa7a15ce672
+	github.com/percona-platform/saas v0.0.0-20230301171036-0eea7770672e
```
Comment (artemgavrilov) [inline go.mod:55]: TODO

File: go.mod [hunk 2]
```diff
@@ -52,7 +52,7 @@ require (
 	github.com/operator-framework/api v0.17.3
 	github.com/operator-framework/operator-lifecycle-manager v0.23.1
 	github.com/percona-platform/dbaas-api v0.0.0-20230103182808-d79c449a9f4c
-	github.com/percona-platform/saas v0.0.0-20221014123257-4fa7a15ce672
+	github.com/percona-platform/saas v0.0.0-20230303105207-ea36b7de2bf1
```
Comment (artemgavrilov) [inline go.mod:55]: TODO

File: managed/cmd/pmm-managed-starlark/main.go [hunk 3]
```diff
@@ -118,20 +117,13 @@ func runChecks(l *logrus.Entry, data *checks.StarlarkScriptData) ([]check.Result
 		return nil, errors.Wrap(err, "error initializing starlark env")
 	}
 
-	res := make([][]map[string]interface{}, len(data.QueriesResults))
-	for i, queryResult := range data.QueriesResults {
-		if res[i], err = agentpb.UnmarshalActionQueryResult(queryResult); err != nil {
-			return nil, err
-		}
-	}
```
Comment (artemgavrilov) [inline managed/cmd/pmm-managed-starlark/main.go:126]: All queries return different result structure, it's become hard to parse and properly unmarshal them here. So I moved unmarhaling to managed, right after we getting action result.

**Comment by artemgavrilov** [comment]
Tests failing because portal part wasn't deployed yet

File: managed/services/checks/checks.go [hunk 4]
```diff
@@ -788,13 +789,13 @@ func (s *Service) executeCheck(ctx context.Context, target services.Target, c ch
 		case check.PostgreSQLShow:
 			eg.Go(func() error {
 				var err error
-				resData[i], err = s.executePostgreSQLShowQuery(gCtx, target)
+				resData[i], err = s.executePostgreSQLShowQuery(ctx, target)
```
Comment (michal-kralik) [inline managed/services/checks/checks.go:792]: Any reason we don't use `gCtx` anymore?

**Review by michal-kralik — APPROVED** [review]

**Review by idoqo — APPROVED** [review]

File: managed/services/checks/checks.go [hunk 5]
```diff
@@ -1222,16 +1249,59 @@ func convertVMValue(value model.Value) ([]byte, error) {
...
 		return nil, errors.WithStack(err)
 	}
 
-	res, err := agentpb.MarshalActionQueryDocsResult(data)
+	return data, nil
+}
+
+func (s *Service) discoverAvailablePGDatabases(ctx context.Context, target services.Target) ([]string, error) {
+	query := check.Query{Query: `datname FROM pg_database  
+WHERE datallowconn = true AND datistemplate = false AND has_database_privilege(current_user, datname, 'connect')`}
```
Comment (PavelKhripkov) [inline managed/services/checks/checks.go:1262]: Can we:
1. Add a link with description of parameters of the query;
2. Move query string to the `const` section?

**Comment by PavelKhripkov** [comment]
Did we implement requirement from the ticket to execute check "on only specified dbs"?
